tests/rados/test_scrub_enhancement.py

```diff
@@ -0,0 +1,204 @@
+            pool_name = config["create_pools"][0]["create_pool"]["pool_name"]
+            log.info("Setting the autoscale mode off")
+            rados_obj.set_pool_property(
+                pool=pool_name, props="pg_autoscale_mode", value="off"
+            )
+            log.info(f"The {pool_name} autoscale mode is off")
+            log.info(f"creating the object inside the {pool_name}")
+            pool_obj.do_rados_put(client=client_node, pool=pool_name, obj_name=oname)
+            log.info(f"A object created inside the {pool_name}")
+            snapcmd = f"sudo rados mksnap -p {pool_name} snap1"
+            client_node.exec_command(cmd=snapcmd, sudo=True)
+            log.info("Creating the SNA_ entry by creating object")
+            pool_obj.do_rados_put(client=client_node, pool=pool_name, obj_name=oname)
+            log.info(f"A snapshot object is created for {pool_name}")
+            osd_map_output = rados_obj.get_osd_map(pool=pool_name, obj=oname)
+            primary_osd = osd_map_output["acting_primary"]
+            log.info(f"The object stored in the primary osd number-{primary_osd}")
+            pg_id = osd_map_output["pgid"]
+            # Stopping OSD
+            if not rados_obj.change_osd_state(action="stop", target=primary_osd):
+                log.error(f"Unable to stop the OSD : {primary_osd}")
+                raise Exception("Execution error")
+            acting_osd_node = rados_obj.fetch_host_node(
+                daemon_type="osd", daemon_id=primary_osd
+            )
+
+            cmd_base = f"cephadm shell --name osd.{primary_osd} --"
+            cmd_get_snaObj = (
+                f"{cmd_base} ceph-kvstore-tool bluestore-kv  /var/lib/ceph/osd/ceph-"
+                f"{primary_osd} list p | grep SNA_"
+            )
+            cmd_no_obj = (
+                f"{cmd_base} ceph-kvstore-tool bluestore-kv  /var/lib/ceph/osd/ceph-{primary_osd} "
+                f"list p | grep SNA_ | wc -l"
+            )
+            # Retiving the snapshot object
+            sna_obj, _ = acting_osd_node.exec_command(sudo=True, cmd=cmd_get_snaObj)
+            sna_obj = sna_obj.strip()
+            sna_obj_split_result = sna_obj.split(".")
+            sna_key_obj_list = sna_obj.split("\t")
+            corrupt_key = sna_obj_split_result[1].strip()
+            # Creating the corrupted key
+            corrupt_key = corrupt_key + "CORRUPTED"
+            # Getting the correct key
+            correct_key = sna_key_obj_list[1].strip()
+            # Creating mount directory in the acting osd to create the object output
+            cmd_create_dir = f"mkdir -p {mount_point}"
+            acting_osd_node.exec_command(sudo=True, cmd=cmd_create_dir)
+            cmd_str_obj = (
+                f"{cmd_base}mount {mount_point}/ -- ceph-kvstore-tool bluestore-kv  /var/lib/ceph/osd/"
+                f"ceph-{primary_osd} get p  {correct_key} out /mnt/output"
+            )
+            # Removing the snapshot object
```

Comment:
I am confused as to how KV-store-tool get p option will remove the snapshot object. Is the comment correct?

Reply:
Got it, we are deleting in next step. It's checking and then removing. Please update the comments.